src/codegate/ca/codegate_ca.py

```diff
@@ -41,194 +108,179 @@ def __init__(self):
-        builder = builder.serial_number(x509.random_serial_number())
-        builder = builder.not_valid_before(datetime.datetime.utcnow())
-        builder = builder.not_valid_after(datetime.datetime.utcnow() + datetime.timedelta(days=365))
-
-        # Add domain to SAN
-        builder = builder.add_extension(
-            x509.SubjectAlternativeName([x509.DNSName(domain)]),
-            critical=False,
-        )
-
-        # Add extended key usage
-        builder = builder.add_extension(
-            x509.ExtendedKeyUsage(
-                [
-                    ExtendedKeyUsageOID.SERVER_AUTH,
-                    ExtendedKeyUsageOID.CLIENT_AUTH,
-                ]
-            ),
-            critical=False,
-        )
-
-        # Basic constraints (not a CA)
-        builder = builder.add_extension(
-            x509.BasicConstraints(ca=False, path_length=None),
-            critical=True,
-        )
-
-        logger.debug(f"Signing certificate for domain: {domain}")
-        certificate = builder.sign(
-            private_key=self._ca_key,
-            algorithm=hashes.SHA256(),
-        )
-
-        # Save domain certificate and key
-        logger.debug(f"Saving certificate and key for domain: {domain}")
-        domain_cert_path = os.path.join(Config.get_config().certs_dir, f"{domain}.crt")
-        domain_key_path = os.path.join(Config.get_config().certs_dir, f"{domain}.key")
-
-        with open(domain_cert_path, "wb") as f:
-            logger.debug(f"Saving certificate to {domain_cert_path}")
-            f.write(certificate.public_bytes(serialization.Encoding.PEM))
 
-        with open(domain_key_path, "wb") as f:
-            logger.debug(f"Saving key to {domain_key_path}")
-            f.write(
-                key.private_bytes(
-                    encoding=serialization.Encoding.PEM,
-                    format=serialization.PrivateFormat.PKCS8,
-                    encryption_algorithm=serialization.NoEncryption(),
-                )
-            )
+                self._ca_last_load_time = current_time
+                logger.debug("Successfully loaded and cached CA certificates")
+                return self._ca_cert, self._ca_key
+
+            except Exception as e:
+                logger.error(f"Failed to load CA certificates: {e}")
+                # Clear cached values on error
+                self._ca_cert = None
+                self._ca_key = None
+                self._ca_cert_expiry = None
+                self._ca_last_load_time = None
+                raise
+
+    def remove_certificates(self) -> None:
+        """Remove all cached certificates and contexts"""
+        logger.debug("Removing all cached certificates and contexts")
+        self.certs_dir = Config.get_config().certs_dir
+        # remove and recreate certs directory
+        try:
+            logger.debug(f"Removing certs directory: {self.certs_dir}")
+            os.rmdir(self.certs_dir)
```

Comment:
This will only work if the certs_dir is empty, right? Or did you mean to use rmtree here?